Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.inl

```diff
@@ -91,23 +91,17 @@ namespace AZ::Dom
     }
 
     template<class T>
-    bool DomPrefixTree<T>::AttachNodeAtPath(const Path& path, Node&& nodeToAttach)
+    void DomPrefixTree<T>::AttachNodeAtPath(const Path& path, Node&& nodeToAttach)
     {
         Node* node = &m_rootNode;
         const size_t entriesToIterate = path.Size() - 1;
         for (size_t i = 0; i < entriesToIterate; ++i)
         {
             const PathEntry& entry = path[i];
-            auto nodeIt = node->m_values.find(entry);
-            if (nodeIt == node->m_values.end())
-            {
-                return false;
-            }
-            node = &nodeIt->second;
+            node = &node->m_values[entry]; // get or create an entry in this node
```

Comment:
Actually I think we should have separate API similar to std::(unordered_)map:: [insert_or_assign](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) function which always updates an entry, but returns a bool indicating whether it is a new entry.

The existing function being able to not attach if there is not an existing sub-tree seems fine